src/task.py

```diff
@@ -7,57 +7,79 @@
 
 class HIIPower(HIITask):
     inputs = {
-        "dmsp_viirs_calibrated": {
+        "nightlights": {
             "ee_type": HIITask.IMAGECOLLECTION,
-            "ee_path": "projects/HII/v1/source/nightlights/dmsp_viirs_calibrated",
+            "ee_path": "nightlights_path",
             "maxage": 1,
-        },
+        }
     }
+
     quantiles = {
-        "0": {"value": 0, "min": 0, "max": 0},
-        "1": {"value": 1, "min": 1, "max": 4},
-        "2": {"value": 2, "min": 5, "max": 5},
-        "3": {"value": 3, "min": 6, "max": 6},
-        "4": {"value": 4, "min": 7, "max": 7},
-        "5": {"value": 5, "min": 8, "max": 9},
-        "6": {"value": 6, "min": 10, "max": 11},
-        "7": {"value": 7, "min": 12, "max": 16},
-        "8": {"value": 8, "min": 17, "max": 30},
-        "9": {"value": 9, "min": 31, "max": 62},
+        "0": {"value": 0, "min": 0, "max": 9},
+        "1": {"value": 1, "min": 10, "max": 10},
+        "2": {"value": 2, "min": 11, "max": 12},
+        "3": {"value": 3, "min": 13, "max": 13},
+        "4": {"value": 4, "min": 14, "max": 16},
+        "5": {"value": 5, "min": 17, "max": 19},
+        "6": {"value": 6, "min": 20, "max": 25},
+        "7": {"value": 7, "min": 26, "max": 35},
+        "8": {"value": 8, "min": 36, "max": 51},
+        "9": {"value": 9, "min": 52, "max": 62},
         "10": {"value": 10, "min": 63, "max": 63},
     }
 
+    thresholds = {
+        "global": 10,
+        "latitude": {"min_lat": 45, "min_val": 10, "max_lat": 60, "max_val": 30},
+    }
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+        self.nightlights, _ = self.get_most_recent_image(
+            ee.ImageCollection(self.inputs["nightlights"]["ee_path"])
+        )
         self.quantiles = ee.Dictionary(self.quantiles)
 
-    @property
-    def nightlights(self):
-        # If nightlights for previous year has already been calculated and stored, use it
-        nightlights_ic = ee.ImageCollection(
-            self.inputs["dmsp_viirs_calibrated"]["ee_path"]
+    def nightlights_path(self):
+        if self.taskdate.year < 2013:
+            return "projects/sat-io/open-datasets/Harmonized_NTL/dmsp"
+        else:
+            return "projects/sat-io/open-datasets/Harmonized_NTL/viirs"
+
+    def latitude_mask(self):
```

Comment:
I think some comments in this function, or perhaps in the repo README (or both), would help -- the code looks great but I don't really know why it's needed or done the way it's done.